src/containers/App.js

```diff
@@ -0,0 +1,47 @@
+import React, { Component } from 'react';
+
+import Header from '../components/Header/Header';
+import Todos from '../components/Todos/Todos';
+import Footer from '../components/Footer/Footer';
+
+class App extends Component {
+  state = {
+    todoList: []
+  }
+
+  onChangeTodoHandler = (item) => {
+    const updatedTodos = this.state.todoList;
+    updatedTodos.push(item);
+    this.setState({todoList: updatedTodos});
+  }
+  
+  onCompletedHandler = (id) => {
+    const updatedTodoIndex = this.state.todoList.findIndex((todo) => todo.id === id);
+    const updatedTodoItem = {...this.state.todoList[updatedTodoIndex]};
```

Comment:
why do you clone the object todo to another object?

Reply:
- We need to treat this.state as if it were immutable. Because mutate this.state directly afterwards may replace mutation we made. The "setState" is work behind the scenes. This means a manual state mutation may be overridden when setState is processed.
- We can't load the difference when we declare a shouldComponentUpdate method. So the 
operator "===" check inside is superfluous because the object reference will not change.